Retrieve and log video information with ffprobe when JAVE fails to process it

The bot sometimes fails to process what should be a valid video (based on its MIME type) with the following exception:

```
com.github.stickerifier.stickerify.telegram.exception.TelegramApiException: ws.schild.jave.InputFormatException
at com.github.stickerifier.stickerify.media.MediaHelper.retrieveMultimediaInfo(MediaHelper.java:321)
at com.github.stickerifier.stickerify.media.MediaHelper.convertToWebm(MediaHelper.java:299)
at com.github.stickerifier.stickerify.media.MediaHelper.convert(MediaHelper.java:65)
at com.github.stickerifier.stickerify.bot.Stickerify.answerFile(Stickerify.java:115)
at com.github.stickerifier.stickerify.bot.Stickerify.answerFile(Stickerify.java:100)
at com.github.stickerifier.stickerify.bot.Stickerify.answer(Stickerify.java:90)
at com.github.stickerifier.stickerify.bot.Stickerify.lambda$handleUpdates$1(Stickerify.java:79)
at java.base/java.util.concurrent.ThreadPerTaskExecutor$TaskRunner.run(Unknown Source)
at java.base/java.lang.VirtualThread.run(Unknown Source)

Caused by: ws.schild.jave.InputFormatException: null
at ws.schild.jave.MultimediaObject.parseMultimediaInfo(MultimediaObject.java:422)
at ws.schild.jave.MultimediaObject.getInfo(MultimediaObject.java:171)
at com.github.stickerifier.stickerify.media.MediaHelper.retrieveMultimediaInfo(MediaHelper.java:319)
... 8 common frames omitted
```

To ease finding the root issue, we can retrieve video's information using `ffprobe` when an exception is thrown in the [method `retrieveMultimediaInfo`](https://github.com/Stickerifier/Stickerify/blob/main/src/main/java/com/github/stickerifier/stickerify/media/MediaHelper.java#L317-L323).

Given this issue comes from the dependency [JAVE 2](https://github.com/a-schild/jave2), we can use the information we will retrieve to report them this _possible_ misbehaviour.

--- a/Dockerfile
+++ b/Dockerfile
@@ -11,5 +11,6 @@ FROM gcr.io/distroless/base-nossl AS bot
 COPY --from=builder /app/build/jre ./jre
 COPY --from=builder /app/build/libs/Stickerify-shadow.jar .
 COPY --from=mwader/static-ffmpeg /ffmpeg /usr/local/bin/
+COPY --from=mwader/static-ffmpeg /ffprobe /usr/local/bin/
 ENV FFMPEG_PATH=/usr/local/bin/ffmpeg
 CMD ["jre/bin/java", "-jar", "Stickerify-shadow.jar"]

--- a/src/main/java/com/github/stickerifier/stickerify/media/MediaHelper.java
+++ b/src/main/java/com/github/stickerifier/stickerify/media/MediaHelper.java
@@ -318,8 +318,29 @@ public final class MediaHelper {
 		try {
 			return new MultimediaObject(file, FFMPEG_LOCATOR).getInfo();
 		} catch (EncoderException e) {
+			logVideoInformation(file);
 			throw new TelegramApiException(e);
 		}
+	}
+
+	/**
+	 * Logs the multimedia information of passed-in video file.
+	 *
+	 * @param file the video to retrieve information from
+	 * @throws TelegramApiException if an error occurred processing the video
+	 */
+	private static void logVideoInformation(File file) throws TelegramApiException {
+		var ffprobeCommand = new String[] {
+				"ffprobe",
+				"-v", "quiet",
+				"-show_format",
+				"-show_streams",
+				file.getAbsolutePath()
+		};
+
+		var videoInformation = ProcessHelper.executeCommand(ffprobeCommand);
+
+		LOGGER.atWarn().log("The video could not be processed successfully\n{}", videoInformation);
 	}
 
 	/**

--- a/src/main/java/com/github/stickerifier/stickerify/media/MediaHelper.java
+++ b/src/main/java/com/github/stickerifier/stickerify/media/MediaHelper.java
@@ -316,6 +316,7 @@ public final class MediaHelper {
 	 */
 	private static MultimediaInfo retrieveMultimediaInfo(File file) throws TelegramApiException {
 		try {
+			logVideoInformation(file);
 			return new MultimediaObject(file, FFMPEG_LOCATOR).getInfo();
 		} catch (EncoderException e) {
 			logVideoInformation(file);

--- a/src/main/java/com/github/stickerifier/stickerify/media/MediaHelper.java
+++ b/src/main/java/com/github/stickerifier/stickerify/media/MediaHelper.java
@@ -316,7 +316,6 @@ public final class MediaHelper {
 	 */
 	private static MultimediaInfo retrieveMultimediaInfo(File file) throws TelegramApiException {
 		try {
-			logVideoInformation(file);
 			return new MultimediaObject(file, FFMPEG_LOCATOR).getInfo();
 		} catch (EncoderException e) {
 			logVideoInformation(file);

--- a/src/main/java/com/github/stickerifier/stickerify/process/ProcessHelper.java
+++ b/src/main/java/com/github/stickerifier/stickerify/process/ProcessHelper.java
@@ -43,7 +43,7 @@ public final class ProcessHelper {
 				throw new TelegramApiException("The command {} couldn't complete {}: {}", command[0], reason, output);
 			}
 
-			return readStream(process.getInputStream());
+			return readProcessOutput(process);
 		} catch (IOException | InterruptedException e) {
 			throw new TelegramApiException(e);
 		} finally {
@@ -63,6 +63,21 @@ public final class ProcessHelper {
 	 */
 	private static String readStream(InputStream stream) throws IOException {
 		return new String(stream.readAllBytes(), UTF_8);
+	}
+
+	/**
+	 * Reads the content of the input stream of the process.
+	 * If the input stream is empty, the content of the error stream is returned.
+	 *
+	 * @param process the process
+	 * @return the output of the process
+	 * @throws IOException if an error occurs reading stream's bytes
+	 */
+	private static String readProcessOutput(Process process) throws IOException {
+		var inputStream = readStream(process.getInputStream());
+		var output = inputStream.isEmpty() ? readStream(process.getErrorStream()) : inputStream;
+
+		return output.trim();
 	}
 
 	private ProcessHelper() {

